tests/functional/tests/management/test_start_stop.py

```diff
@@ -0,0 +1,471 @@
+    Check that OCF does not allow for 2 caches to be started with the same cache_id
+    """
+
+    cache_device1 = Volume(Size.from_MiB(20))
+    cache_device2 = Volume(Size.from_MiB(20))
+    cache_id = randrange(1, 16385)
+    cache = Cache.start_on_device(cache_device1, cache_mode=mode, cache_line_size=cls, cache_id=cache_id)
+    cache.get_stats()
+
+    with pytest.raises(OcfError, match="OCF_ERR_CACHE_EXIST"):
+        cache = Cache.start_on_device(cache_device2, cache_mode=mode, cache_line_size=cls, cache_id=cache_id)
+    cache.get_stats()
+
+
+@pytest.mark.parametrize("mode", CacheMode)
+@pytest.mark.parametrize("cls", CacheLineSize)
+def test_start_cache_same_device(pyocf_ctx, mode, cls):
+    """Adding two caches using the same cache device
+    Check that OCF does not allow for 2 caches using the same cache device to be started
+    """
+
+    cache_device = Volume(Size.from_MiB(20))
+    cache = Cache.start_on_device(cache_device, cache_mode=mode, cache_line_size=cls)
+    cache.get_stats()
+
+    with pytest.raises(OcfError, match="OCF_ERR_NOT_OPEN_EXC"):
+        cache = Cache.start_on_device(cache_device, cache_mode=mode, cache_line_size=cls)
+    cache.get_stats()
+
+
+@pytest.mark.parametrize("mode", CacheMode)
+@pytest.mark.parametrize("cls", CacheLineSize)
+def test_start_too_small_device(pyocf_ctx, mode, cls):
+    """Starting cache with device below 100MiB
+    Check if starting cache with device below minimum size is blocked
+    """
+
+    cache_device = Volume(Size.from_B(20 * 1024 * 1024 - 1))
+
+    with pytest.raises(OcfError, match="OCF_ERR_START_CACHE_FAIL"):
+        Cache.start_on_device(cache_device, cache_mode=mode, cache_line_size=cls)
+
+
+def run_io_and_cache_data_if_possible(exported_obj, mode, cls):
+    test_data = Data.from_string("This is test data")
+
+    if mode in {CacheMode.WI, CacheMode.WA}:
+        logger.info("[STAGE] Write to core device")
+        io_to_core(exported_obj, test_data, 20, True)
+        logger.info("[STAGE] Read from exported object")
+        io_from_exported_object(exported_obj, test_data.size, 20)
+    else:
+        logger.info("[STAGE] Write to exported object")
+        io_to_core(exported_obj, test_data, 20)
+
+    stats = exported_obj.cache.get_stats()
+    logger.info("[ASSERT] Occupancy")
+    assert stats["usage"]["occupancy"]["value"] == \
+        ((cls / CacheLineSize.LINE_4KiB) if mode != CacheMode.PT else 0)
+
+
+def io_to_core(exported_obj: Core, data: Data, offset: int, to_core_device=False):
+    io = exported_obj.new_core_io() if to_core_device else exported_obj.new_io()
+    io.set_data(data)
+    io.configure(offset, data.size, IoDir.WRITE, 0, 0)
+    io.set_queue(exported_obj.cache.get_default_queue())
+
+    completion = OcfCompletion([("err", c_int)])
+    io.callback = completion.callback
+    io.submit()
+    completion.wait()
+
+    logger.info("[ASSERT] IO to exported object completion")
+    assert completion.results["err"] == 0
```

Comment:
Assert supports syntax which includes explanation for it, this way you don't have to write to logs and have it show up nicely in pytest fail logs:
`assert <condition>, "Message for the assert"`
Picked a random assert for this comment, but it applies to (nearly?) all asserts in this file

Reply:
Done